Makefile

```diff
@@ -9,10 +9,8 @@ PROJ = google_containers
 
 PKG_SOURCES := $(shell find pkg -name '*.go')
 
-node-problem-detector: ./bin/node-problem-detector
-
-./bin/node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
-	CGO_ENABLED=0 GOOS=linux godep go build -a -installsuffix cgo -ldflags '-w' -o node-problem-detector
+node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
+	GOOS=linux go build -ldflags '-w -extldflags "-static"' -o node-problem-detector
```

Comment:
> My kernel-monitor configuration is here.

@andyxning Yeah, with your configuration, we'll not run go-systemd related code I believe.

> I guess that once we make node-problem-detector a statically linked binary, the so file for systemd is also linked into it. :)

Nope. go-systemd explicitly call dlopen to load shared binary. I don't think the compiler understands what happens https://github.com/kubernetes/node-problem-detector/blob/master/vendor/github.com/coreos/go-systemd/sdjournal/functions.go#L46.